inspectors/view/qml/notation/fretdiagrams/FretCanvas.qml

```diff
@@ -0,0 +1,179 @@
+import QtQuick 2.0
+import QtQuick.Controls 2.2
+import "../../common/"
+
+FocusableItem {
+    id: root
+    height: 300
+    width: parent.width
+
+    property QtObject model: null
+
+    property int strings: model ? model.strings.value : 6
+    property int frets: model ? model.frets.value : 4
+    property int fretOffset: model ? model.offset.value : 0
+    property bool showNut: model ? model.showNut.value : true
+    property int numPos: model ? model.numPos.value : 0 // TODO: make this have effect
+    property bool antialiasing: true
+
+    readonly property real mag: 1.5
+    readonly property real spatium: 20.0 * mag
+    readonly property real lineWidth1: spatium * 0.08
+    property real lineWidth2: (fretOffset || !showNut) ? lineWidth1 : spatium * 0.2
+    readonly property real stringDist: spatium * 0.7
+    readonly property real fretDist: spatium * 0.8
+    readonly property real dotD: stringDist * 0.6 + lineWidth1
+
+    property real diagramWidth: (strings - 1) * stringDist
+    property real diagramHeight: (frets * fretDist) + fretDist * 0.5
+
+    Component.onCompleted: {
+        model.onSelectionChanged.connect(frame.requestPaint);
+        model.onSelectionChanged.connect(canvas.requestPaint);
+    }
+
+    FontLoader {
+        id: freeSans
+        source: "../../../../../fonts/FreeSans.ttf"
+    }
+
+    readonly property string basicFont: "%1px %2"
+
+    Canvas {
```

Comment:
I don't quite get you, if you're talking about the properties above, these are what we use in the old fret canvas, and since they're only used in this canvas, I see no need to share them with the model.